src/app/Components/sign-up/sign-up.component.ts

```diff
@@ -11,14 +11,39 @@ import { IUser, CognitoService } from '../../Services/cognito.service';
 export class SignUpComponent {
   loading: boolean;
   user: IUser;
+  public errorInSignUp: string = '';
+  public errorMessage: string = '';
 
   constructor(private router: Router, private cognitoService: CognitoService) {
     this.loading = false;
     this.user = {} as IUser;
   }
 
-  public signUp(): void {
-    this.loading = true;
+  passwordValidate(password: string): void {
+    const regexPassword = /^[\S]+.*[\S]+$/;
+    let validatePassword = (<HTMLInputElement>(
+      document.getElementById('confirmPassword')
```

Comment:
since we are using angular, it would be better to use two way binding that angular provides
change `id=confirmPassword` in html to `[(ngModel)]="confirmPassword"` and declare a class variable `confirmPassword:string = ''`